cmd/portable-package.rb

```diff
@@ -33,15 +33,18 @@ def portable_package
     raise UsageError, "--no-commit requires --write!" if !Homebrew.args.write? && Homebrew.args.no_commit?
 
     ENV["HOMEBREW_DEVELOPER"] = "1"
-    ENV["HOMEBREW_BUILD_BOTTLE"] = "1" if OS.linux?
 
     Homebrew.args.named.each do |name|
       name = "portable-#{name}" unless name.start_with? "portable-"
       begin
         deps = Utils.popen_read("brew", "deps", "--include-build", name).split("\n")
 
         # Avoid installing glibc. Bottles depend on glibc.
-        safe_system "brew", "install", "-s", *deps if OS.linux?
+        if OS.linux?
+          deps.each do |dep|
+            safe_system "brew", "install", "--build-bottle", dep
+          end
```

Comment:
If split like this, is there not ordering issues where you might build a dependent before its dependencies?

Reply:
That feels likely. Maybe this is a sign we really do need that environment variable.

Reply:
I'm 99% sure that `brew deps` returns the dependencies in topological order, that is to say, the order in which you need to install the dependencies.

Reply:
Ah, I think we need to pass `-n` to `brew deps`:

```
-n                               Sort dependencies in topological order.
```

Though I was hinting about making it a single `install *deps` call where I reckon the ordering might not matter. Or theoretically shouldn't if `FormulaInstaller` is smart enough.

Reply:
I believe there are standing issues with dependencies not installing as bottles when the main requested formula is a bottle, but `install *deps` may or may not have that issue. I'll check.